packages/legend-graph/src/models/protocols/pure/v1/engine/V1_Engine.ts

```diff
@@ -471,12 +471,17 @@ export class V1_Engine {
     generationMode: GenerationMode,
     model: V1_PureModelContextData,
   ): Promise<V1_GenerationOutput[]> {
+    const grammar = this.pureModelContextDataToPureCode(model);
+    // TODO: change this to send PureModelContextText
```

Comment:
Like I mentioned, since we seem to have time, let's actually implement this `TODO`

Reply:
Now construct and send PMCT instead of doing the roundtrip back to PMCD from grammar.